### PR TITLE
[ML] Job in Index: Convert get calendar events to index docs

### PR DESCRIPTION
**Pull request by davidkyle**
When a job Id is passed in the request look up groups and jobs from the index documents

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java [hunk 1]
```diff
@@ -631,6 +631,44 @@ public void expandGroupIds(List<String> groupIds,  ActionListener<Set<String>> l
...
+     * @param listener Returns true or failure
+     */
+    public void groupExists(String groupId, ActionListener<Boolean> listener) {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        boolQueryBuilder.filter(new TermQueryBuilder(Job.JOB_TYPE.getPreferredName(), Job.ANOMALY_DETECTOR_JOB_TYPE));
+        boolQueryBuilder.filter(new TermsQueryBuilder(Job.GROUPS.getPreferredName(), groupId));
+
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
+                .query(boolQueryBuilder);
+        sourceBuilder.fetchSource(false);
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java]: I think you should make `.size(0)` on the search as we don't really care about any of the hits, just the total number of them.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java [hunk 2]
```diff
@@ -631,6 +631,44 @@ public void expandGroupIds(List<String> groupIds,  ActionListener<Set<String>> l
...
 
+    /**
+     * Check if a group exists, that is there exists a job that is a member of
+     * the group. If there are one or more jobs that define the group then
+     * the listener responds with true else an error.
+     *
+     * @param groupId The group Id
+     * @param listener Returns true or failure
+     */
+    public void groupExists(String groupId, ActionListener<Boolean> listener) {
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java]: I think that for consistency with `jobExists` the listener should get `Boolean.FALSE` if the group does not exist and it is up to the caller to know if they want to error on `false` or not. Throwing an error here does not seem right as no error actually occurred.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java (hunk 1 above)
Comment (davidkyle) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java]: 👍

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java (hunk 2 above)
Comment (davidkyle) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java]: Yes that provides a better interface for users I pushed the change.

**Review by benwtrent — APPROVED** [review]
LGTM!!!!
🚀

**Pull request merged**